File: Trabalho_prático_02/TP02Q02 - Registro em C/Q02.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <math.h>


typedef struct Serie{

    char nome[50];
    char formato[50];
    char duracao[50];
    char paisDeOrigem[50];
    char idiomaOriginal[50];
    char emissora[50];
    char transmissaoOriginal[50];
    int numTeporadas;
    int numEpisodios;

}Serie;


//construtores
void Construtor1 (Serie *s){
    strcpy(s -> nome, "");
    strcpy(s -> formato, "");
    strcpy(s -> duracao, "");
    strcpy(s -> paisDeOrigem, "");
    strcpy(s -> idiomaOriginal, "");
    strcpy(s -> emissora, "");
    strcpy(s -> transmissaoOriginal, "");
    s -> numTeporadas = 0;
    s -> numEpisodios = 0;
}

void Construtor2 (Serie *s, char nome[], char formato[], char duracao[], char paisDeOrigem[], char idiomaOriginal[], char emissora[], char transmissaoOriginal[], int numTeporadas, int numEpisodios){
    strcpy(s -> nome, nome);
    strcpy(s -> formato, formato);
    strcpy(s -> duracao, duracao);
    strcpy(s -> paisDeOrigem, paisDeOrigem);
    strcpy(s -> idiomaOriginal, idiomaOriginal);
    strcpy(s -> emissora, emissora);
    strcpy(s -> transmissaoOriginal, transmissaoOriginal);
    s -> numTeporadas = numTeporadas;
    s -> numEpisodios = numEpisodios;
}


//setters
void setNome(Serie *s, char nome[]){
    strcpy(s -> nome, nome);
}

void setFormato(Serie *s, char formato[]){
    strcpy(s -> formato, formato);
}

void setDuracao(Serie *s, char duracao[]){
    strcpy(s -> duracao, duracao);
}

void setPaisDeOrigem(Serie *s, char paisDeOrigem[]){
    strcpy(s -> paisDeOrigem, paisDeOrigem);
}

void setIdiomaOriginal(Serie *s, char idiomaOriginal[]){
    strcpy(s -> idiomaOriginal, idiomaOriginal);
}

void setEmissora(Serie *s, char emissora[]){
    strcpy(s -> emissora, emissora);
}

void setTransmissaoOriginal(Serie *s, char transmissaoOriginal[]){
    strcpy(s -> transmissaoOriginal, transmissaoOriginal);
}

void setnumTeporadas(Serie *s, int numTeporadas){
    s -> numTeporadas = numTeporadas;
}

void setNumEpisodios(Serie *s, int numEpisodios){
    s -> numEpisodios = numEpisodios;
}


//getters
char* getNome(Serie s){
    char string[100] = {};
    char *resp = string;
    strcpy(resp, s.nome);
    return resp;
}

char* getFormato(Serie s){
    char string[100] = {};
    char *resp = string;
    strcpy(resp, s.formato);
    return resp;
}

char* getDuracao(Serie s){
    char string[100] = {};
    char *resp = string;
    strcpy(resp, s.duracao);
    return resp;
}

char* getPaisDeOrigem(Serie s){
    char string[100] = {};
    char *resp = string;
    strcpy(resp, s.paisDeOrigem);
    return resp;
}

char* getIdiomaOriginal(Serie s){
    char string[100] = {};
    char *resp = string;
    strcpy(resp, s.idiomaOriginal);
    return resp;
}

char* getEmissora(Serie s){
    char string[100] = {};
    char *resp = string;
    strcpy(resp, s.emissora);
    return resp;
}

char* getTransmissaoOriginal(Serie s){
    char string[100] = {};
    char *resp = string;
    strcpy(resp, s.transmissaoOriginal);
    return resp;
}

int getNumEpisodios(Serie s){
    return s.numEpisodios;
}

int getNumTeporadas(Serie s){
    return s.numTeporadas;
}


//clone 
Serie Clone(Serie s){
    Serie resp;

    strcpy(resp.nome, s.nome);
    strcpy(resp.formato, s.formato);
    strcpy(resp.duracao, s.duracao);
    strcpy(resp.paisDeOrigem, s.paisDeOrigem);
    strcpy(resp.idiomaOriginal, s.idiomaOriginal);
    strcpy(resp.emissora, s.emissora);
    strcpy(resp.transmissaoOriginal, s.transmissaoOriginal);
    resp.numEpisodios = s.numEpisodios;
    resp.numTeporadas = s.numTeporadas;

    return resp;
}


//retira as tags de uma string
char* tiraTags(char *linha){
    char string[100] = {};
    char *novaLinha = string;

    int i=0, j=0;

    while(i < strlen(linha)){
        if(linha[i] == '<'){
            i++;
            while(linha[i] != '>'){
                i++;
            }
        }else{
            novaLinha[j] = linha[i];
            j++;
        }
        i++;
    }

    novaLinha[j] = '\0';

    return novaLinha;
}

//transforma um numero em formato String para formato int
    int transformaEmInteiro(char s[]){
        int num = 0;
        double mult = 0;
        if(strlen(s)>=1){
            num += ((int) s[strlen(s)-1]) - 48;
        }
        if(strlen(s)>=2){
            num += (((int) s[strlen(s)-2]) - 48) * 10;
        }
        if(strlen(s)>=3){
            num += (((int) s[strlen(s)-3]) - 48) * 100;
        }
        if(strlen(s)>=4){
            num += (((int) s[strlen(s)-4]) - 48) * 1000;
        }

        return num;
    }

//retira todos os caracteres nao numeros de uma string
    char* soDeixaNumeros(char s[]){
        char string[100] = {};
        char *novaLinha = string;
        int j = 0;

        for(int i=0; i<strlen(s); i++){
            if(s[i]>='0' && s[i]<='9'){
                novaLinha[j] = s[i];
                j++;
            }
        }

        novaLinha[j] = '\0';
        return novaLinha;
    }

//retira espacos no inicio ou final da string
    char* trim(char s[]) {
        char string[100] = {};
        char *novaLinha = string;
        int posInicio=0, posFinal=strlen(s)-1;
        int j=0;

        for(int i=0; i<strlen(s); i++){
            if(s[i] != ' '){
                posInicio = i;
                i = strlen(s);
            }
        }
        
        for(int i=strlen(s)-1; i>0; i--){
            if(s[i]!=' '){
                posFinal = i;
                i = 0;
            }
        }

        for(int i=posInicio; i<=posFinal; i++){
            novaLinha[j] = s[i];
            j++;
        }
    
        novaLinha[j] = '\0';
        return novaLinha;
    }

//retira informacoes disnecessarias como "&#160;"
    char* retiraSpan(char s[]) {
        char string[100] = {};
        char *novaLinha = string;
        int j=0;

        for(int i=0; i<strlen(s); i++){
            if(s[i]=='&'){
                i++;
                while(s[i]!=';'){
                    i++;
                }
            }else{
                novaLinha[j] = s[i];
                j++;
            }
        }

        novaLinha[j] = '\0';
        return novaLinha;
    }

//formata o nome da serie
    char* formataNome(char nome[]){
        char string[100] = {};
        char *novoNome = string;
        int j=0;

        for(int i=0; i<strlen(nome); i++){
            if(nome[i]=='_'){
                novoNome[j] = ' ';
                j++;
            }else if(nome[i]=='.' && nome[i+1]=='h' && nome[i+2]=='t' && nome[i+3]=='m' && nome[i+4]=='l'){
                i += 5;
            }else{
                novoNome[j] += nome[i];
                j++;
            }
        }

        novoNome[j] = '\0';
        return novoNome;
    }

//retira quebras de linha de uma string
    char* retiraQuebraDeLinha(char s[]){
        char string[100] = {};
        char *novaLinha = string;
        int j=0;

        for(int i=0; i<strlen(s); i++){
            if(s[i]=='\n'){
                i++;
            }else{
                novaLinha[j] = s[i];
                j++;
            }
        }

        novaLinha[j] = '\0';
        return novaLinha;

    }


//metodo ler
void ler(Serie *s, char entrada[]){
    
    char arquivo[100] = "/tmp/series/";
    strcat(arquivo, entrada);
    
    FILE *arq = fopen(arquivo, "r");
        char aux[200];
        char aux2[200];


        char buf[2000]; 
        fgets(buf, 2000, arq);

        //nome
        setNome(s, formataNome(entrada));

        //formato
        while(strstr(buf, "Formato") == NULL){
            fgets(buf, 2000, arq);
        }
        fgets(buf, 2000, arq);
        setFormato(s, tiraTags(buf));

        //Duração
        while(strstr(buf, "Duração") == NULL){
            fgets(buf, 2000, arq);
        }
        fgets(buf, 2000, arq);
        setDuracao(s, tiraTags(buf));

        //País de origem
        while(strstr(buf, "País de origem") == NULL){
            fgets(buf, 2000, arq);
        }
        fgets(buf, 2000, arq);
        strcpy(aux, tiraTags(buf));
        strcpy(aux2, trim(aux));
        setPaisDeOrigem(s, retiraSpan(aux2));

        //Idioma original
        while(strstr(buf, "Idioma original") == NULL){
            fgets(buf, 2000, arq);
        }
        fgets(buf, 2000, arq);
        strcpy(aux, tiraTags(buf));
        setIdiomaOriginal(s, trim(aux));

        //Emissora
        while(strstr(buf, "Emissora") == NULL){
            fgets(buf, 2000, arq);
        }
        fgets(buf, 2000, arq);
        strcpy(aux,tiraTags(buf) );
        setEmissora(s, trim(aux));
        

        //Transmissão original
        while(strstr(buf, "Transmissão original") == NULL){
            fgets(buf, 2000, arq);
        }
        fgets(buf, 2000, arq);
        strcpy(aux, tiraTags(buf));
        strcpy(aux2, trim(aux));
        setTransmissaoOriginal(s, retiraSpan(aux2));

        //temporadas
        while(strstr(buf, "temporadas") == NULL){
            fgets(buf, 2000, arq);
        }
        fgets(buf, 2000, arq);
        strcpy(aux, tiraTags(buf));
        strcpy(aux2, soDeixaNumeros(aux));
        setnumTeporadas(s, transformaEmInteiro(aux2));

        //episódios
        while(strstr(buf, "episódios") == NULL){
            fgets(buf, 2000, arq);
        }
        fgets(buf, 2000, arq);
        strcpy(aux, tiraTags(buf));
        strcpy(aux2, soDeixaNumeros(aux));
        setNumEpisodios(s, transformaEmInteiro(aux2));


    fclose(arq);
}


//metodo imprimir
void imprimir(Serie s){
    printf("%s ", retiraQuebraDeLinha(s.nome));
    printf("%s ", retiraQuebraDeLinha(s.formato));
    printf("%s ", retiraQuebraDeLinha(s.duracao));
    printf("%s ", retiraQuebraDeLinha(s.paisDeOrigem));
    printf("%s ", retiraQuebraDeLinha(s.idiomaOriginal));
    printf("%s ", retiraQuebraDeLinha(s.emissora));
    printf("%s ", retiraQuebraDeLinha(s.transmissaoOriginal));
    printf("%i ", s.numTeporadas);
    printf("%i\n", s.numEpisodios);
}




//checa a condicao de parada
int isFim(char string[]){
    if(strlen(string)==3 && string[0]=='F' && string[1]=='I' && string[2]=='M'){
        return 1;
    }else{
        return 0;
    }
}

int main (void){

    char entrada[100];
    scanf("%[^\n]s", entrada);
    scanf("%*[^\n]s"); scanf("%*c");

    Serie serie;
    int i=0;

    while( isFim(entrada) == 0){
        Construtor1(&serie);

        ler(&serie, entrada);

        imprimir(serie);

        scanf("%[^\n]s", entrada);
        scanf("%*[^\n]s"); scanf("%*c");
        
        i++;
    }
    
}

File: Trabalho_prático_04/TP04Q02 - Arvore Arvore/Q02.java
import java.io.BufferedReader;
import java.io.FileInputStream;
import java.io.InputStreamReader;
import java.io.*;
import java.util.Date;

class Serie{

    // declaracao de atributos
    private String nome;
    private String formato;
    private String duracao;
    private String paisDeOrigem;
    private String idiomaOriginal;
    private String emissora;
    private String transmissaoOriginal;
    private int numTeporadas;
    private int numEpisodios;

    //construtores
    Serie(){
        nome = "";
        formato = "";
        duracao = "";
        paisDeOrigem = "";
        idiomaOriginal = "";
        emissora = "";
        transmissaoOriginal = "";
        numTeporadas = 0;
        numEpisodios = 0;
    }

    Serie(String nome, String formato, String duracao, String paisDeOrigem, String idiomaOriginal, String emissora, String transmissaoOriginal, int numTeporadas, int numEpisodios){
        nome = this.nome;
        formato = this.formato;
        duracao = this.duracao;
        paisDeOrigem = this.paisDeOrigem;
        idiomaOriginal = this.idiomaOriginal;
        emissora = this.emissora;
        transmissaoOriginal = this.transmissaoOriginal;
        numTeporadas = this.numTeporadas;
        numEpisodios = this.numEpisodios;
    }

    //setters e getters
    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getFormato() {
        return formato;
    }

    public void setFormato(String formato) {
        this.formato = formato;
    }

    public String getDuracao() {
        return duracao;
    }

    public void setDuracao(String duracao) {
        this.duracao = duracao; 
    }

    public String getPaisDeOrigem() {
        return paisDeOrigem;
    }

    public void setPaisDeOrigem(String paisDeOrigem) {
        this.paisDeOrigem = paisDeOrigem;
    }

    public String getIdiomaOriginal() {
        return idiomaOriginal;
    }

    public void setIdiomaOriginal(String idiomaOriginal) {
        this.idiomaOriginal = idiomaOriginal;
    }

    public String getEmissora() {
        return emissora;
    }

    public void setEmissora(String emissora) {
        this.emissora = emissora;
    }

    public String getTransmissaoOriginal() {
        return transmissaoOriginal;
    }

    public void setTransmissaoOriginal(String transmissaoOriginal) {
        this.transmissaoOriginal = transmissaoOriginal;
    }

    public int getNumTeporadas() {
        return numTeporadas;
    }

    public void setNumTeporadas(int numTeporadas) {
        this.numTeporadas = numTeporadas;
    }

    public int getNumEpisodios() {
        return numEpisodios;
    }

    public void setNumEpisodios(int numEpisodios) {
        this.numEpisodios = numEpisodios;
    }

    //metodos clone
    public Serie clone(){
        Serie resp = new Serie();

        resp.nome = this.nome;
        resp.formato = this.formato;
        resp.duracao = this.duracao;
        resp.paisDeOrigem = this.paisDeOrigem;
        resp.idiomaOriginal = this.idiomaOriginal;
        resp.emissora = this.emissora;
        resp.transmissaoOriginal = this.transmissaoOriginal;
        resp.numTeporadas = this.numTeporadas;
        resp.numEpisodios = this.numEpisodios;

        return resp;
    }

    //metodo imprimir
    public void imprimir(){
        System.out.println(nome + " " + formato + " " + duracao + " " + paisDeOrigem + " " + idiomaOriginal + " " + emissora + " " + transmissaoOriginal + " " + numTeporadas + " " + numEpisodios);
    }

    //retira as tags de uma string
    public String tiraTags(String html) {
        String semTags = "";
        int i=0;

        while(i<html.length()){
            if(html.charAt(i)=='<'){
                i++;
                while(html.charAt(i)!='>'){
                    i++;
                }
            }else{
                semTags += html.charAt(i);
            }
            i++;
        }

        return semTags;
    }

    //transforma um numero em formato String para formato int
    public int transformaEmInteiro(String s){
        int num = 0;
        int mult = 0;

        for(int i=s.length()-1; i>=0; i--){
            num += (((int) s.charAt(i)) - 48) * Math.pow(10, mult);
            mult++;
        }

        return num;
    }

    //retira todos os caracteres nao numeros de uma string
    public String soDeixaNumeros(String s){
        String novaLinha="";

        for(int i=0; i<s.length(); i++){
            if(s.charAt(i)==' '){
                i = s.length();
            }else{
                novaLinha += s.charAt(i);
            }
        }

        return novaLinha;
    }

    //retira espacos no inicio ou final da string
    public String trim(String s) {
        String novaLinha="";
        int posInicio=0, posFinal=s.length()-1;

        for(int i=0; i<s.length(); i++){
            if(s.charAt(i)!=' '){
                posInicio = i;
                i = s.length();
            }
        }
        
        for(int i=s.length()-1; i>0; i--){
            if(s.charAt(i)!=' '){
                posFinal = i;
                i = 0;
            }
        }

        for(int i=posInicio; i<=posFinal; i++){
            novaLinha += s.charAt(i);
        }
    
        return novaLinha;
    }

    //retira informacoes disnecessarias como "&#160;" da transmissao original
    public String retiraSpan(String s) {
        String novaLinha="";

        for(int i=0; i<s.length(); i++){
            if(s.charAt(i)=='&'){
                i++;
                while(s.charAt(i)!=';'){
                    i++;
                }
            }else{
                novaLinha += s.charAt(i);
            }
        }

        return novaLinha;
    }

    //formata o nome da serie
    public String formataNome(String nome){
        String novoNome="";

        for(int i=0; i<nome.length(); i++){
            if(nome.charAt(i)=='_'){
                novoNome += ' ';
            }else if(nome.charAt(i)=='/'){
                i += 11;
                

            }else if(nome.charAt(i)=='.' && nome.charAt(i+1)=='h' && nome.charAt(i+2)=='t' && nome.charAt(i+3)=='m' && nome.charAt(i+4)=='l'){
                i += 5;
            }else{
                novoNome += nome.charAt(i);
            }
        }

        return novoNome;
    }

    //metodo ler
    public void ler(String nomeArquivo) throws Exception{
        InputStreamReader isr = new InputStreamReader(new FileInputStream(nomeArquivo));
        BufferedReader br = new BufferedReader(isr);
        
        this.nome = formataNome(nomeArquivo);

        while(!br.readLine().contains("Formato"));
        this.formato = tiraTags(br.readLine());

        while(!br.readLine().contains("Duração")); 
        this.duracao = tiraTags(br.readLine());

        while(!br.readLine().contains("País de origem"));
        this.paisDeOrigem = trim(retiraSpan(tiraTags(br.readLine())));

        while(!br.readLine().contains("Idioma original"));
        this.idiomaOriginal = trim(tiraTags(br.readLine()));

        while(!br.readLine().contains("Emissora"));
        this.emissora = trim(tiraTags(br.readLine()));

        while(!br.readLine().contains("Transmissão original"));
        this.transmissaoOriginal = trim(retiraSpan(tiraTags(br.readLine())));

        while(!br.readLine().contains("temporadas"));
        this.numTeporadas = transformaEmInteiro(soDeixaNumeros(tiraTags(br.readLine())));

        while(!br.readLine().contains("episódios"));
        this.numEpisodios = Integer.parseInt(soDeixaNumeros(tiraTags(br.readLine())));

        br.close();
    }

}



class No {
	public char elemento; // Conteudo do no.
	public No esq; // No da esquerda.
	public No dir; // No da direita.
   public No2 outro;
	
   /**
	 * Construtor da classe.
	 * @param elemento Conteudo do no.
	 */
	No(char elemento) {
		this.elemento = elemento;
		this.esq = this.dir = null;
      this.outro = null;
	}

	/**
	 * Construtor da classe.
	 * @param elemento Conteudo do no.
	 * @param esq No da esquerda.
	 * @param dir No da direita.
	 */
	No(char elemento, No esq, No dir) {
		this.elemento = elemento;
		this.esq = esq;
		this.dir = dir;
      this.outro = null;
	}
}



class No2 {
	public String elemento; // Conteudo do no.
	public No2 esq; // No da esquerda.
	public No2 dir; // No da direita.
	
   /**
	 * Construtor da classe.
	 * @param elemento Conteudo do no.
	 */
	No2(String elemento) {
		this.elemento = elemento;
		this.esq = this.dir = null;
	}

	/**
	 * Construtor da classe.
	 * @param elemento Conteudo do no.
	 * @param esq No2 da esquerda.
	 * @param dir No2 da direita.
	 */
	No2(String elemento, No2 esq, No2 dir) {
		this.elemento = elemento;
		this.esq = esq;
		this.dir = dir;
	}
}



class ArvoreArvore {
	private No raiz; // Raiz da arvore.

    // Construtor da classe
    public ArvoreArvore() {
        raiz = null;
        try {
            inserir('D');
        } catch (Exception e) {}
        try {
            inserir('R');
        } catch (Exception e) {}
        try {
            inserir('Z');
        } catch (Exception e) {}  
        try {
            inserir('X');
        } catch (Exception e) {}
        try {
            inserir('V');
        } catch (Exception e) {}
        try {
            inserir('B');
        } catch (Exception e) {}
        try {
            inserir('F');
        } catch (Exception e) {}
        try {
            inserir('P');
        } catch (Exception e) {}
        try {
            inserir('U');
        } catch (Exception e) {}
        try {
            inserir('I');
        } catch (Exception e) {}
        try {
            inserir('G');
        } catch (Exception e) {}
        try {
            inserir('E');
        } catch (Exception e) {}
        try {
            inserir('J');
        } catch (Exception e) {}
        try {
            inserir('L');
        } catch (Exception e) {}
        try {
            inserir('H');
        } catch (Exception e) {}
        try {
            inserir('T');
        } catch (Exception e) {}
        try {
            inserir('A');
        } catch (Exception e) {}
        try {
            inserir('W');
        } catch (Exception e) {}
        try {
            inserir('S');
        } catch (Exception e) {}
        try {
            inserir('O');
        } catch (Exception e) {}
        try {
            inserir('M');
        } catch (Exception e) {}
        try {
            inserir('N');
        } catch (Exception e) {}
        try {
            inserir('K');
        } catch (Exception e) {}
        try {
            inserir('C');
        } catch (Exception e) {}
        try {
            inserir('Y');
        } catch (Exception e) {}
        try {
            inserir('Q');
        } catch (Exception e) {}
    }

    //inserir letra
    public void inserir(char letra) throws Exception {
        raiz = inserir(letra, raiz);
    }

    private No inserir(char x, No i) throws Exception {
        if (i == null) {
            i = new No(x);

        } else if (x < i.elemento) {
            i.esq = inserir(x, i.esq);

        } else if (x > i.elemento) {
            i.dir = inserir(x, i.dir);

        }

        return i;
	}

    // Inserir string
    public void inserir(String s, Log log) throws Exception{
        inserir(s, raiz, log);
    }

    public void inserir(String s, No i, Log log) throws Exception {
        if (i == null) {
            throw new Exception("Erro ao inserir: caractere invalido!");

        } else if (s.charAt(0) < i.elemento) {
            log.comp++;
            inserir(s, i.esq, log);
            

        } else if (s.charAt(0) > i.elemento) {
            log.comp++;
            log.comp++;
            inserir(s, i.dir, log);

        } else {
            log.comp++;
            log.comp++;
            i.outro = inserir(s, i.outro, log);
        }
    }

    private No2 inserir(String s, No2 i, Log log) throws Exception {
        if (i == null) {
            i = new No2(s);

        } else if (s.compareTo(i.elemento) < 0) {
            log.comp++;
            i.esq = inserir(s, i.esq, log);

        } else if (s.compareTo(i.elemento) > 0) {
            log.comp++;
            log.comp++;
            i.dir = inserir(s, i.dir, log);

        } else {
            throw new Exception("Erro ao inserir: elemento existente!");
        }

        return i;
    }


    // Mostrar o caminho e pesquisar elementos
    public boolean mostrar(String x, Log log){
        return mostrar(raiz, x, log);
    }

    public boolean mostrar(No i, String x, Log log){
        boolean resp = false;
        if (i != null){
            if(mostrar(i.outro, x, log) == true){
                resp = true;
                return resp;
            }
            if(resp == false){
                MyIO.print("esq ");
                resp = mostrar(i.esq, x, log);

                if(resp == false){
                    MyIO.print("dir ");
                    resp = mostrar(i.dir, x, log);

                }
            }
        }
        return resp;
    }

    public boolean mostrar(No2 i, String x, Log log){
        boolean resp = false;
        if (i != null){
            if(x.compareTo(i.elemento) == 0){
                log.comp++;
                resp = true;
                return resp;
            }
            if(resp == false){
                log.comp++;
                MyIO.print("ESQ ");
                resp = mostrar(i.esq, x, log);

                if(resp == false){
                    MyIO.print("DIR ");
                    resp = mostrar(i.dir, x, log);

                }
            }
        }
    
        return resp;
    }


}



class Log{
    public int comp;
}

public class Q02 {

    public static boolean isFIM(String s) {
        return(s.length()==3 && s.charAt(0)=='F' && s.charAt(1)=='I' && s.charAt(2)=='M');
    }



    public static void main(String[] args) {
        long inicio = new Date().getTime(); //pega a hora exata de inicio
        String entrada = MyIO.readLine();
        Serie serie = new Serie();
        ArvoreArvore arvore = new ArvoreArvore();
        Log log = new Log();
        
        // inserir elementos na arvore
        while(isFIM(entrada)==false){
            
            try {
                serie.ler("/tmp/series/" + entrada);
            } catch (Exception e) {}
            
            try {
                arvore.inserir(serie.getNome(), log);
            } catch (Exception e) {}
            
            serie = new Serie();
            entrada = MyIO.readLine();

        }

        // realizar pesquisas
        entrada = MyIO.readLine();
        
        while(isFIM(entrada) == false ){
            MyIO.print("raiz ");
            if(arvore.mostrar(entrada, log) == true){
                MyIO.print(" SIM\n");
            }else{
                MyIO.print(" NAO\n");
            }
            
            entrada = MyIO.readLine();
        }


        long fim = new Date().getTime(); //pega a hora exata do fim
        double tempo = (fim-inicio)/1000; //calcula o tempo de execucao
        int numComparacoes = log.comp;

        //preenchimento do arquivo log
        Arq.openWrite(" matricula arvoreArvore.txt", "UTF-8");
        Arq.print("741019\t"); //matricula
        Arq.print(tempo+"\t"); //tempo de execucao
        Arq.print(numComparacoes+"\t"); // comparacoe
        Arq.close();

    }

}
